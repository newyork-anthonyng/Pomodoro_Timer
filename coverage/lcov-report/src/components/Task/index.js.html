<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Task/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/Task</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/8</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-yes">21x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import T from 'prop-types';
import noop from '../../utility/noop';
import glamorous from 'glamorous';
&nbsp;
const H4 = glamorous.h4({
  display: 'inline-block',
  margin: 0,
});
&nbsp;
const P = glamorous.p({
  marginTop: '10px',
});
&nbsp;
const Button = glamorous.button({
  float: 'right',
});
&nbsp;
const Task = ({ id, title, text, onDeleteClick }) =&gt; (
  &lt;div data-taskId={id}&gt;
    &lt;H4&gt;{title}&lt;/H4&gt;
&nbsp;
    &lt;Button onClick={() =&gt; onDeleteClick(id)}&gt;Ã—&lt;/Button&gt;
&nbsp;
    &lt;P&gt;{text}&lt;/P&gt;
  &lt;/div&gt;
);
&nbsp;
Task.defaultProps = {
  onDeleteClick: noop,
};
&nbsp;
Task.propTypes = {
  id: T.string.isRequired,
  title: T.string.isRequired,
  text: T.string.isRequired,
  onDeleteClick: T.func.isRequired,
};
&nbsp;
export default Task;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat May 06 2017 15:20:25 GMT-0700 (PDT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
