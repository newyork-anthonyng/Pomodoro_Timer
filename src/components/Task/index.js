import React from 'react';
import T from 'prop-types';
import noop from '../../utility/noop';
import glamorous from 'glamorous';

const H4 = glamorous.h4({
  display: 'inline-block',
  margin: 0,
});

const P = glamorous.p({
  marginTop: '10px',
});

const Button = glamorous.button({
  float: 'right',
});

const Task = ({ id, title, text, onDeleteClick }) => (
  <div data-taskId={id}>
    <H4>{title}</H4>

    <Button onClick={() => onDeleteClick(id)}>Ã—</Button>

    <P>{text}</P>
  </div>
);

Task.defaultProps = {
  onDeleteClick: noop,
};

Task.propTypes = {
  id: T.string.isRequired,
  title: T.string.isRequired,
  text: T.string.isRequired,
  onDeleteClick: T.func.isRequired,
};

export default Task;
