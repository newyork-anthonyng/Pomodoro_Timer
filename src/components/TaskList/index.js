import React from 'react';
import T from 'prop-types';
import Task from '../Task';
import noop from '../../utility/noop';
import glamorous from 'glamorous';

const Container = glamorous.div({
  marginTop: '40px',
  width: '250px',
});

const TaskList = ({ tasks, onDeleteClick }) => (
  <Container>
    {
      tasks.map((task) => (
        <Task
          {...task}
          key={task.id}
          onDeleteClick={onDeleteClick}
        />
      ))
    }
  </Container>
);

TaskList.defaultProps = {
  tasks: [],
  onDeleteClick: noop,
};

TaskList.propTypes = {
  tasks: T.arrayOf(T.shape({
    id: T.string.isRequired,
    title: T.string.isRequired,
    text: T.string.isRequired,
  })).isRequired,
  onDeleteClick: T.func.isRequired,
};

export default TaskList;
