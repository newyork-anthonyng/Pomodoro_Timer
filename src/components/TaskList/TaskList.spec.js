import React from 'react';
import { shallow } from 'enzyme';
import TaskList from './';
import Task from '../Task';
import noop from '../../utility/noop';

const defaultProps = {
  tasks: [
    { id: 'batmanId', title: 'Batman', text: 'The Dark Knight' },
    { id: 'supermanId', title: 'Superman', text: 'Wears red underwear' },
  ],
  onDeleteClick: noop,
};

it('should render one component per task', () => {
  const wrapper = shallow(
    <TaskList {...defaultProps} />
  );

  expect(wrapper.find(Task).length).toEqual(2);
  // snapshot test for each Task component
  for (let i = 0, sum = wrapper.find(Task).length; i < sum; i++) {
    expect(wrapper.find(Task).at(i)).toMatchSnapshot();
  }
});
