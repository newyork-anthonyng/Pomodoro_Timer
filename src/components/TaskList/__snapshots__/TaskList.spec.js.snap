// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render one component per task 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Task
    id="batmanId"
    onDeleteClick={[Function]}
    text="The Dark Knight"
    title="Batman"
/>,
  "nodes": Array [
    <Task
      id="batmanId"
      onDeleteClick={[Function]}
      text="The Dark Knight"
      title="Batman"
/>,
  ],
  "options": Object {},
  "renderer": null,
  "root": ShallowWrapper {
    "complexSelector": ComplexSelector {
      "buildPredicate": [Function],
      "childrenOfNode": [Function],
      "findWhereUnwrapped": [Function],
    },
    "length": 1,
    "node": <glamorous(div)>
      <Task
            id="batmanId"
            onDeleteClick={[Function]}
            text="The Dark Knight"
            title="Batman"
      />
      <Task
            id="supermanId"
            onDeleteClick={[Function]}
            text="Wears red underwear"
            title="Superman"
      />
</glamorous(div)>,
    "nodes": Array [
      <glamorous(div)>
        <Task
                id="batmanId"
                onDeleteClick={[Function]}
                text="The Dark Knight"
                title="Batman"
        />
        <Task
                id="supermanId"
                onDeleteClick={[Function]}
                text="Wears red underwear"
                title="Superman"
        />
</glamorous(div)>,
    ],
    "options": Object {},
    "renderer": ReactShallowRenderer {
      "_instance": ShallowComponentWrapper {
        "_calledComponentWillUnmount": false,
        "_compositeType": 2,
        "_context": Object {},
        "_currentElement": <TaskList
          onDeleteClick={[Function]}
          tasks={
                    Array [
                              Object {
                                "id": "batmanId",
                                "text": "The Dark Knight",
                                "title": "Batman",
                              },
                              Object {
                                "id": "supermanId",
                                "text": "Wears red underwear",
                                "title": "Superman",
                              },
                            ]
          }
/>,
        "_debugID": 1,
        "_hostContainerInfo": null,
        "_hostParent": null,
        "_instance": StatelessComponent {
          "_reactInternalInstance": [Circular],
          "context": Object {},
          "props": Object {
            "onDeleteClick": [Function],
            "tasks": Array [
              Object {
                "id": "batmanId",
                "text": "The Dark Knight",
                "title": "Batman",
              },
              Object {
                "id": "supermanId",
                "text": "Wears red underwear",
                "title": "Superman",
              },
            ],
          },
          "refs": Object {},
          "state": null,
          "updater": Object {
            "enqueueCallback": [Function],
            "enqueueCallbackInternal": [Function],
            "enqueueElementInternal": [Function],
            "enqueueForceUpdate": [Function],
            "enqueueReplaceState": [Function],
            "enqueueSetState": [Function],
            "isMounted": [Function],
            "validateCallback": [Function],
          },
        },
        "_mountOrder": 1,
        "_pendingCallbacks": null,
        "_pendingElement": null,
        "_pendingForceUpdate": false,
        "_pendingReplaceState": false,
        "_pendingStateQueue": null,
        "_renderedComponent": NoopInternalComponent {
          "_currentElement": <glamorous(div)>
            <Task
                        id="batmanId"
                        onDeleteClick={[Function]}
                        text="The Dark Knight"
                        title="Batman"
            />
            <Task
                        id="supermanId"
                        onDeleteClick={[Function]}
                        text="Wears red underwear"
                        title="Superman"
            />
</glamorous(div)>,
          "_debugID": 2,
          "_renderedOutput": <glamorous(div)>
            <Task
                        id="batmanId"
                        onDeleteClick={[Function]}
                        text="The Dark Knight"
                        title="Batman"
            />
            <Task
                        id="supermanId"
                        onDeleteClick={[Function]}
                        text="Wears red underwear"
                        title="Superman"
            />
</glamorous(div)>,
        },
        "_renderedNodeType": 1,
        "_rootNodeID": 0,
        "_topLevelWrapper": null,
        "_updateBatchNumber": null,
        "_warnedAboutRefsInRender": false,
      },
      "getRenderOutput": [Function],
      "render": [Function],
    },
    "root": [Circular],
    "unrendered": <TaskList
      onDeleteClick={[Function]}
      tasks={
            Array [
                  Object {
                    "id": "batmanId",
                    "text": "The Dark Knight",
                    "title": "Batman",
                  },
                  Object {
                    "id": "supermanId",
                    "text": "Wears red underwear",
                    "title": "Superman",
                  },
                ]
      }
/>,
  },
  "unrendered": null,
}
`;

exports[`should render one component per task 2`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Task
    id="supermanId"
    onDeleteClick={[Function]}
    text="Wears red underwear"
    title="Superman"
/>,
  "nodes": Array [
    <Task
      id="supermanId"
      onDeleteClick={[Function]}
      text="Wears red underwear"
      title="Superman"
/>,
  ],
  "options": Object {},
  "renderer": null,
  "root": ShallowWrapper {
    "complexSelector": ComplexSelector {
      "buildPredicate": [Function],
      "childrenOfNode": [Function],
      "findWhereUnwrapped": [Function],
    },
    "length": 1,
    "node": <glamorous(div)>
      <Task
            id="batmanId"
            onDeleteClick={[Function]}
            text="The Dark Knight"
            title="Batman"
      />
      <Task
            id="supermanId"
            onDeleteClick={[Function]}
            text="Wears red underwear"
            title="Superman"
      />
</glamorous(div)>,
    "nodes": Array [
      <glamorous(div)>
        <Task
                id="batmanId"
                onDeleteClick={[Function]}
                text="The Dark Knight"
                title="Batman"
        />
        <Task
                id="supermanId"
                onDeleteClick={[Function]}
                text="Wears red underwear"
                title="Superman"
        />
</glamorous(div)>,
    ],
    "options": Object {},
    "renderer": ReactShallowRenderer {
      "_instance": ShallowComponentWrapper {
        "_calledComponentWillUnmount": false,
        "_compositeType": 2,
        "_context": Object {},
        "_currentElement": <TaskList
          onDeleteClick={[Function]}
          tasks={
                    Array [
                              Object {
                                "id": "batmanId",
                                "text": "The Dark Knight",
                                "title": "Batman",
                              },
                              Object {
                                "id": "supermanId",
                                "text": "Wears red underwear",
                                "title": "Superman",
                              },
                            ]
          }
/>,
        "_debugID": 1,
        "_hostContainerInfo": null,
        "_hostParent": null,
        "_instance": StatelessComponent {
          "_reactInternalInstance": [Circular],
          "context": Object {},
          "props": Object {
            "onDeleteClick": [Function],
            "tasks": Array [
              Object {
                "id": "batmanId",
                "text": "The Dark Knight",
                "title": "Batman",
              },
              Object {
                "id": "supermanId",
                "text": "Wears red underwear",
                "title": "Superman",
              },
            ],
          },
          "refs": Object {},
          "state": null,
          "updater": Object {
            "enqueueCallback": [Function],
            "enqueueCallbackInternal": [Function],
            "enqueueElementInternal": [Function],
            "enqueueForceUpdate": [Function],
            "enqueueReplaceState": [Function],
            "enqueueSetState": [Function],
            "isMounted": [Function],
            "validateCallback": [Function],
          },
        },
        "_mountOrder": 1,
        "_pendingCallbacks": null,
        "_pendingElement": null,
        "_pendingForceUpdate": false,
        "_pendingReplaceState": false,
        "_pendingStateQueue": null,
        "_renderedComponent": NoopInternalComponent {
          "_currentElement": <glamorous(div)>
            <Task
                        id="batmanId"
                        onDeleteClick={[Function]}
                        text="The Dark Knight"
                        title="Batman"
            />
            <Task
                        id="supermanId"
                        onDeleteClick={[Function]}
                        text="Wears red underwear"
                        title="Superman"
            />
</glamorous(div)>,
          "_debugID": 2,
          "_renderedOutput": <glamorous(div)>
            <Task
                        id="batmanId"
                        onDeleteClick={[Function]}
                        text="The Dark Knight"
                        title="Batman"
            />
            <Task
                        id="supermanId"
                        onDeleteClick={[Function]}
                        text="Wears red underwear"
                        title="Superman"
            />
</glamorous(div)>,
        },
        "_renderedNodeType": 1,
        "_rootNodeID": 0,
        "_topLevelWrapper": null,
        "_updateBatchNumber": null,
        "_warnedAboutRefsInRender": false,
      },
      "getRenderOutput": [Function],
      "render": [Function],
    },
    "root": [Circular],
    "unrendered": <TaskList
      onDeleteClick={[Function]}
      tasks={
            Array [
                  Object {
                    "id": "batmanId",
                    "text": "The Dark Knight",
                    "title": "Batman",
                  },
                  Object {
                    "id": "supermanId",
                    "text": "Wears red underwear",
                    "title": "Superman",
                  },
                ]
      }
/>,
  },
  "unrendered": null,
}
`;
